packages/use-sync-external-store/src/__tests__/useSyncExternalStoreShimServer-test.js

```diff
@@ -0,0 +1,98 @@
+/**
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @emails react-core
+ *
+ * @jest-environment node
+ */
+
+'use strict';
+
+let useSyncExternalStore;
+let React;
+let ReactDOM;
+let ReactDOMServer;
+let Scheduler;
+
+// This tests the userspace shim of `useSyncExternalStore` in a server-rendering
+// (Node) environment
+describe('useSyncExternalStore (userspace shim, server rendering)', () => {
+  beforeEach(() => {
+    jest.resetModules();
+
+    // Remove useSyncExternalStore from the React imports so that we use the
+    // shim instead. Also removing startTransition, since we use that to detect
+    // outdated 18 alphas that don't yet include useSyncExternalStore.
+    //
+    // Longer term, we'll probably test this branch using an actual build of
+    // React 17.
+    jest.mock('react', () => {
+      const {
+        // eslint-disable-next-line no-unused-vars
+        startTransition: _,
```

Comment:
just curious why do you need to remove these fields for the test?

Reply:
It's to simulate a build of React 17, which includes neither `useSyncExternalStore` nor `startTransition`.

The reason `startTransition` is relevant is because I use it as a heuristic to check whether you're using an outdated alpha of React 18: https://github.com/facebook/react/blob/main/packages/use-sync-external-store/src/useSyncExternalStore.js#L49-L61

The eventual plan is to load a copy of React 17 from npm and test the shim using that.